### PR TITLE
Garden Pages

### PR DESCRIPTION
**Pull request by hashmaster3k**
# Garden Pages

Go get the basic skeletal structure of the webpages. API calls are not made in this PR.

### ```GET /dashboard ``` - dashboard
Displays the user's gardens, if any. Otherwise has a call to action (CTA) to create a new garden. The add garden button remains when new gardens are added. With each displayed garden is a static image, along with an edit button and trash button. The two said buttons are also links to edit and delete that garden. Note: The images are utilizing the asset pipeline. However, due to not having oAuth established yet, I was unable to get the images to display with just using save_and_open_page as running ```rails s``` is not an option.

### ```GET /gardens/new``` - new garden page
The user has four input fields: name, latitude, longitude, description and two radio buttons: yes or no (for private garden). By default the no button is checked, so it is up to the user to specify if they want their garden to be "hidden" (see future extension on social platform). Once submitted, hits the ```post /gardens```.

### ```GET /gardens/:id``` - garden show page
If a user has a garden, the image on their dashboard is a link to that specific garden. They can click on the image which will route them to that gardens show page. On that page will be information regarding that garden, it's plants, it's sensors and buttons to add them if none exist. There will also be a weather information block. This is currently empty as we are waiting for the micro-service and back-end team to make this available.

### ```GET /gardens/:id/edit``` - edit garden page
From the dashboard a user can click the edit garden icon which will redirect them to a form similar to garden creation. The garden information is currently being created into a PORO in the controller. Once the back-end API is working fully, we will utilize the facade, service system to create the PORO and pass the information onto the view.

### ```POST /gardens``` - create new garden
The controller currently does not hit the back-end API. Waiting on back-end to finalize needed columns, serializers, etc. The user ID is also going to be passed into the params hash via HTTP request to the API so the back-end knows which garden this will belong to.

### ```PATCH /garden/:id```
Controller currently does not hit the back-end API. Only redirected for testing.

### ```DELETE /garden/:id```
Controller currently does not hit the back-end API. Only redirected for testing.

## Additional Notes

Currently all tests are passing with 99.37% coverage. However, tests are not robust. Further testing will be needed once calls to BE API is implemented.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -14,7 +14,7 @@ gem 'sass-rails', '~> 5.0'
 # Use Uglifier as compressor for JavaScript assets
 gem 'uglifier', '>= 1.3.0'
 # See https://github.com/rails/execjs#readme for more supported runtimes
-# gem 'mini_racer', platforms: :ruby
+gem 'mini_racer', platforms: :ruby
```
Comment (hashmaster3k) [inline Gemfile:17]: Enabled this gem to allow Travis-CI build to successfully build.

File: app/views/gardens/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<center>
+  <article class='left-col'>
+    <section class='garden-info'>
+      <h1><%= @garden.name %></h1>
```
Comment (dcoleman21) [inline app/views/gardens/show.html.erb:4]: @leahriffell add location and description; static hardcode an image

**Review by dcoleman21 — COMMENTED** [review]
Looks fabulous @hashmaster3k

**Review by LHJE — COMMENTED** [review]
Very impressive!

File: app/views/gardens/show.html.erb (hunk 2 above)
Comment (leahriffell) [reply app/views/gardens/show.html.erb:4]: @dcoleman21 done!

File: config/routes.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
   get '/privacy', to: 'privacy#index'
+  
+  get '/dashboard', to: 'dashboard#show'
+  
+  resources :gardens, expect: :index
```
Comment (leahriffell) [inline config/routes.rb:7]: @hashmaster3k should this be 'except'? I can change it on my branch if so.

**Pull request merged**